README.md

```diff
@@ -6,40 +6,89 @@ This repository contains an example function, that is compatible with Speckle's
+
+This sets up a CLI application with two commands:
 
 1. the main function command, that implements the Speckle Automate function's anatomy
-2. a helper command that can generate the JSON Schema from the function author provided `FunctionInputs` class. This command is called whenever a new version of the function is published to automate.
+2. `generate-schema` -> a helper command that can generate the JSON Schema from the function author provided `FunctionInputs` class. This command is called whenever a new version of the function is published to automate.
 
-### Function functionality - AutomateFunction.cs
+### AutomateFunction.cs - The function implementation
 
 The `AutomateFunction.cs` contains the actual function implementation. This is the file that you should modify to implement the function's functionality.
-The `Run` function takes the parsed arguments that match the 3 arguments schema described [above](#functional-anatomy).
 
-## Github Action
+The `Run` function takes 2 inputs:
+
+- `AutomationContext` -> The context of your automation, which contains all the parsed information provided by the automate service as explained [above](#anatomy-of-a-function)
+- **Optional** A `struct` representing your desired input data (see [Function Inputs](#functioninputscs---the-function-inputs))
+
+The template already contains an example implementation that will count how many objects of a particular type can be found on a version.
+
+### FunctionInputs.cs - The function inputs
+
+The definition of the user-defined inputs required for the function to work. This will also be used by the `generate-schema` CLI command to inform Automate of the required inputs a user setting up an automation will need to provide.
+
+This is just a `struct` with whatever properties you need. For now, we support basic types (string, int, double) and nested `struct`s with the same type limitations as its parent.
+
+## Publishing a function
 
 We publish a [github action](https://github.com/specklesystems/speckle-automate-github-composite-action) that bundles all the steps that are needed to publish a new version of a function. The automation requires two secret values, the `SPECKLE_FUNCTION_ID` and `SPECKLE_FUNCTION_TOKEN` both of these are provided, when you register a new function on Automate.
 
+This will:
+
+- Restore any packages and build your function
+- Generate the json schema for your inputs
+- Package your function as a Docker image
+- Register the new image as a new version in Speckle Automate
+
+Once this process has successfully finished, your function should be available and discoverable in Speckle.Automate
+
 > [!NOTE]
-> You need to register your function on Speckle Automate to get the required secret values.
+> You need to register your function on Speckle Automate to get the required secret values. For new functions, this is automatically done by our GitHub Application, so no extra user setup is needed.
```

Comment:
Register your function on Speckle Automate to get the needed secret values. This is automatically done for new functions by our GitHub Application, so no extra setup is required on your end.